mlflow/server/js/src/gateway/utils/gatewayUtils.ts

```diff
@@ -0,0 +1,44 @@
+import type { Endpoint } from '../types';
+
+export const GATEWAY_MODEL_PREFIX = 'gateway:/';
+
+export enum ModelProvider {
+  GATEWAY = 'gateway',
+  OTHER = 'other',
+}
+
+export const getModelProvider = (model: string | undefined): ModelProvider => {
+  if (!model || model.startsWith(GATEWAY_MODEL_PREFIX)) {
+    return ModelProvider.GATEWAY;
+  }
+  return ModelProvider.OTHER;
+};
+
+export const getEndpointNameFromModel = (model: string | undefined): string | undefined => {
+  if (model?.startsWith(GATEWAY_MODEL_PREFIX)) {
+    return model.replace(GATEWAY_MODEL_PREFIX, '');
+  }
+  return undefined;
+};
+
+export const formatModelFromEndpoint = (endpointName: string): string => {
+  return `${GATEWAY_MODEL_PREFIX}${endpointName}`;
+};
+
+export const isDirectModel = (model: string | undefined): boolean => {
+  return Boolean(model && !model.startsWith(GATEWAY_MODEL_PREFIX));
+};
```

Comment:
IIUC effectively this function checks if it'a a non-gateway model, can we add a JSdoc for that?